### PR TITLE
add user api frontend

### PR DESCRIPTION
**Pull request by bohdan-koldun**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,3 +43,18 @@ To start storybook:
...
+### User api frontend (GRUD) frontend/src/services
+
+```
+import {userAPI} from './[relative path]/services'
+```
+``` userAPI.fetchUser(id) ``` - get user by id
+
+``` userAPI.addUser(user)``` - add 'user'
+
+```userAPI.updateUser(user)``` - update 'user' data by 'user.id'
```
Comment (Narudir) [inline README.md]: missed a space

File: logview/frontend/src/services/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import axios, { AxiosInstance } from 'axios';
```
Comment (Narudir) [inline logview/frontend/src/services/adapter.ts]: AxiosInstance is not used anywhere, you can remove it

File: logview/frontend/src/services/domains/user.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import api from '../adapter';
+
+export default {
+	addUser: (user: any) => {
+		return api.makeRequest(`/api/users`, api.requestType.POST, user);
+	},
+	updateUser: (user: any) => {
+		return api.makeRequest(
+			`/api/users/${user._id}`,
+			api.requestType.PATCH,
```
Comment (Narudir) [inline logview/frontend/src/services/domains/user.ts]: I think it's better to use PUT here:
The main difference between the PUT and PATCH method is that the PUT method uses the request URI to supply a modified version of the requested resource which replaces the original version of the resource whereas the PATCH method supplies a set of instructions to modify the resource.

File: logview/frontend/src/services/index.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import user from './domains/user';
+
+export const userAPI = user;
+
+// here you can add other API
```
Comment (Narudir) [inline logview/frontend/src/services/index.ts]: you can remove this comment

File: logview/frontend/src/services/domains/user.ts (hunk 3 above)
Comment (bohdan-koldun) [reply logview/frontend/src/services/domains/user.ts]: fixed

File: logview/frontend/src/services/adapter.ts (hunk 2 above)
Comment (bohdan-koldun) [reply logview/frontend/src/services/adapter.ts]: fixed

File: README.md (hunk 1 above)
Comment (bohdan-koldun) [reply README.md]: fixed

File: logview/frontend/src/services/index.ts (hunk 4 above)
Comment (bohdan-koldun) [reply logview/frontend/src/services/index.ts]: fixed

**Comment by bohdan-koldun** [comment]
close

**Pull request closed without merging**